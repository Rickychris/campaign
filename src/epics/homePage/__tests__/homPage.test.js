import React from "react";
import ReactDom from "react-dom";
import { BrowserRouter } from "react-router-dom";
import { render, cleanup } from "@testing-library/react";
import HomePage from "../homePage";

describe("HomePage tests", () => {
  beforeEach(cleanup);
  const HomePageTest = (
    <BrowserRouter>
      <HomePage />
    </BrowserRouter>
  );
  it("renders the HomePage", () => {
    const div = document.createElement("div");
    ReactDom.render(HomePageTest, div);
  });

  it("renders the HomePage with text correctly", () => {
    const { getByTestId } = render(HomePageTest);
    expect(getByTestId("homepage")).toHaveTextContent(
      "Create Your Email Marketing Campaign Today!!"
    );
  });

  it("matches snapshot", () => {
    const { getByTestId } = render(HomePageTest);
    expect(getByTestId("homepage")).toMatchSnapshot();
  });
});
