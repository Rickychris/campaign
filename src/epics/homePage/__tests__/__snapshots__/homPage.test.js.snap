// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HomePage tests matches snapshot 1`] = `
<div
  class="root"
  data-testid="homepage"
>
  <div
    class="placeholder"
  >
    Create Your Email Marketing
    <br />
     Campaign Today!!
  </div>
  <div
    class="campaign-wrap"
  >
    <a
      href="/create-campaign"
    >
      <button
        class="root"
        data-testid="button"
        text="Create Campaign"
      >
        Create Campaign
      </button>
    </a>
    <div
      class="root"
      data-testid="spacer"
      style="height: 40px;"
    />
    <a
      href="/view-campaign"
    >
      <button
        class="root"
        data-testid="button"
        text="Manage Campaigns"
      >
        Manage Campaigns
      </button>
    </a>
  </div>
</div>
`;
